services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py

```diff
@@ -70,58 +148,89 @@ async def get_study_job(
 
 @router.delete(
     "/{study_id:uuid}/jobs/{job_id:uuid}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    responses={status.HTTP_404_NOT_FOUND: {"model": ErrorGet}},
     include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,
-    status_code=status.HTTP_501_NOT_IMPLEMENTED,
-    response_description="Not implemented",
 )
-async def delete_study_job(study_id: StudyID, job_id: JobID):
-    msg = f"delete study job study_id={study_id!r} job_id={job_id!r}.  SEE https://github.com/ITISFoundation/osparc-simcore/issues/4111"
-    raise NotImplementedError(msg)
+async def delete_study_job(
+    study_id: StudyID,
+    job_id: JobID,
+    webserver_api: Annotated[AuthSession, Depends(get_webserver_session)],
+):
+    """Deletes an existing study job"""
+    job_name = _compose_job_resource_name(study_id, job_id)
+    _logger.debug("Deleting Job '%s'", job_name)
+    await webserver_api.delete_project(project_id=job_id)
 
 
 @router.post(
     "/{study_id:uuid}/jobs/{job_id:uuid}:start",
     response_model=JobStatus,
     include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,
-    status_code=status.HTTP_501_NOT_IMPLEMENTED,
-    response_description="Not implemented",
 )
 async def start_study_job(
+    request: Request,
     study_id: StudyID,
     job_id: JobID,
-):
-    msg = f"start study job study_id={study_id!r} job_id={job_id!r}. SEE https://github.com/ITISFoundation/osparc-simcore/issues/4177"
-    raise NotImplementedError(msg)
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+    webserver_api: Annotated[AuthSession, Depends(get_webserver_session)],
+    director2_api: Annotated[DirectorV2Api, Depends(get_api_client(DirectorV2Api))],
+    cluster_id: ClusterID | None = None,
+) -> JobStatus:
+    job_name = _compose_job_resource_name(study_id, job_id)
+    _logger.debug("Starting Job '%s'", job_name)
+
+    await start_project(
+        request=request,
+        job_id=job_id,
+        expected_job_name=job_name,
+        webserver_api=webserver_api,
+        cluster_id=cluster_id,
+    )
+    return await inspect_study_job(
+        study_id=study_id,
+        job_id=job_id,
+        user_id=user_id,
+        director2_api=director2_api,
+    )
 
 
 @router.post(
     "/{study_id:uuid}/jobs/{job_id:uuid}:stop",
-    response_model=Job,
+    response_model=JobStatus,
     include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,
-    status_code=status.HTTP_501_NOT_IMPLEMENTED,
-    response_description="Not implemented",
 )
 async def stop_study_job(
     study_id: StudyID,
     job_id: JobID,
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+    director2_api: Annotated[DirectorV2Api, Depends(get_api_client(DirectorV2Api))],
 ):
-    msg = f"stop study job study_id={study_id!r} job_id={job_id!r}. SEE https://github.com/ITISFoundation/osparc-simcore/issues/4177"
-    raise NotImplementedError(msg)
+    job_name = _compose_job_resource_name(study_id, job_id)
+    _logger.debug("Stopping Job '%s'", job_name)
```

Comment:
TIP: we could use `log_context` for all these?

Reply:
Yep, makes sense. I will adapt. Thanks